go.mod

```diff
@@ -251,4 +251,4 @@ replace (
 	k8s.io/klog/v2 => github.com/simonpasquier/klog-gokit/v2 v2.0.1
 )
 
-go 1.17
+go 1.18
```

Comment:
Do we actually use any features specific to 1.18 in the code base?
If it's not the case better not to upgrade this.

Reply:
I don't think we're using any 1.18 feature. I put this back down to `1.17` in https://github.com/thanos-io/thanos/pull/5487/commits/1b79b42b97bbea4494655c5815ada8de266d3e03